### DISCUSSION TITLE
Setting a field to undefined during reset prevents the field from updating

### DISCUSSION THREAD

**Discussion by bezreyhan** [react-hook-form/react-hook-form#5393] [Q&A] [answered]
**Describe the bug**
Here is the relevant code:
```
type FormValues = {
  firstName: { value: string };
  lastName: { value: string };
};

reset({
        firstName: apiResponse.firstName
          ? { value: apiResponse.firstName.foo }
          : undefined,
        lastName: apiResponse.lastName
          ? { value: apiResponse.lastName.bar }
          : undefined
      });
```

If `reset` is called with `undefined` (e.g. `reset({firstName: undefined}))` then that field will always remain as `undefined` even if the user updates the input.

**To Reproduce**

In the CSB: 
1. press the toggle button
2. fill out the "lastName" input
3. submit the form
4. look inside of the console to see that lastName is still undefined.

**Codesandbox link (Required)**
https://codesandbox.io/s/react-hook-form-reset-to-undefined-h0j1b?file=/src/App.tsx

**Expected behavior**
What I would expect is that if a field is set to undefined it is as if if was never set.

**Screenshots**
Here is a demo of the issue via CSB: 

https://user-images.githubusercontent.com/6013763/120024817-988f4500-bfa4-11eb-8099-28c8b2c64590.mov

**Notes**
There are definitely ways for me to work around this issue but they do require more code and are slightly less typesafe. I can recursively strip out all undefined values before I call reset.

Also, sorry if this is expected behavior. I did read that if an input is disabled it will return `undefined` but I did not think that it would disable the field if I manually set it to undefined. Setting the field to null also caused this issue for me. In short, it would be nice if there was a way to set a field as null or undefined w/o disabling the field.

Thanks for your time and help

**Comment by bluebill1049** [comment, marked as answer]
we don't support fields to be set to `undefined`, you will have to use null or empty string instead.

**Reply by bezreyhan** [reply]
@bluebill1049 the issue also persists with `null`. I updated the CSB to use `null` instead of `undefined`: https://codesandbox.io/s/react-hook-form-reset-to-undefined-h0j1b?file=/src/App.tsx


https://user-images.githubusercontent.com/6013763/120221104-a9cd9100-c1f2-11eb-83bc-f7906d71d6e8.mov

**Reply by bezreyhan** [reply]
@bluebill1049 just wanted to follow up on this. The problem still persists if the value is set to `null`.

**Reply by bluebill1049** [reply]
have you tried the latest version?

**Reply by bezreyhan** [reply]
Just tried it out and it is working perfectly. Thanks for the update!
